pkg/controller/infrastructure/actuator_helper.go

```diff
@@ -25,10 +25,14 @@ func patchProviderStatusAndState(
 	infra *extensionsv1alpha1.Infrastructure,
 	status *v1alpha1.InfrastructureStatus,
 	state *runtime.RawExtension,
+	egressCidrs *[]string,
```

Comment:
Passing egressCidrs as pointer slice is imp not necessary.
Just pass an empty slice if it's not set.